liota/dccs/iotcc.py

```diff
@@ -102,31 +115,35 @@ def register(self, entity_obj):
         else:
             # finally will create a RegisteredEntity
             log.info("Registering resource with IoTCC {0}".format(entity_obj.name))
+            reg_resp_q = Queue.Queue()
 
-            def on_response(msg):
+            def on_response(msg, reg_resp_q):
                 try:
                     log.debug("Received msg: {0}".format(msg))
                     json_msg = json.loads(msg)
                     log.debug("Processing msg: {0}".format(json_msg["type"]))
                     self._check_version(json_msg)
-                    if json_msg["type"] == "UNSUPPORTED_VERSION":
-                         raise Exception("Exception in registering resource, version mismatch. Response received from server:" + json_msg["body"])
                     if json_msg["type"] == "create_or_find_resource_response" and json_msg["body"]["uuid"] != "null" and \
                                     json_msg["body"]["id"] == entity_obj.entity_id:
                         log.info("FOUND RESOURCE: {0}".format(json_msg["body"]["uuid"]))
                         self.reg_entity_id = json_msg["body"]["uuid"]
                     else:
                         log.info("Waiting for resource creation")
-                        on_response(self.recv_msg_queue.get(True, timeout))
+                        on_response(reg_resp_q.get(True, timeout), reg_resp_q)
                 except Exception as err:
                     log.exception("Exception while registering resource")
                     raise err
 
             if entity_obj.entity_type == "EdgeSystem":
                 entity_obj.entity_type = "HelixGateway"
+            transaction_id = self._next_id()
+            req = Request(transaction_id, reg_resp_q)
+            log.debug("Updating resource registration queue for transaction_id:{0}".format(transaction_id))
+            with self.req_ops_lock:
+                self.req_dict.update({transaction_id: req})
```

Comment:
The implementation is kept similar to the changes in v2 where request object is used to hold Guid or can hold additional request parameters as well in future (alert_id in case of alert generation request). Hence, kept the implementation same.